### PR TITLE
Update operator-sdk to 1.2.0

### PR DESCRIPTION
**Pull request by jpkrohling**
Signed-off-by: Juraci Paixão Kröhling <juraci@kroehling.de>

### CODE REVIEW

**Comment by jpkrohling** [comment]
cc @rubenvp9510 would you be able to review this one?

**Comment by rubenvp8510** [comment]
LGTM

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -88,3 +88,58 @@ Every bug fix should be accompanied with a unit test, so that we can prevent reg
...
 
 They are mostly welcome!
+
+## Operator SDK
+
+Coming soon, but for the moment:
+
+Build the operator
+```
+BUNDLE_VERSION=0.16.0
```
Comment (objectiser) [inline CONTRIBUTING.md:98]: Isn't this 0.17.1 now?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -88,3 +88,58 @@ Every bug fix should be accompanied with a unit test, so that we can prevent reg
...
+make set-image-controller IMG=quay.io/jpkroehling/opentelemetry-operator
```
Comment (objectiser) [inline CONTRIBUTING.md:99]: would it be better to replace your user id with something that obviously needs to be replaced? Or above set `export USER_ID=.....`  and then use `${USER_ID}` in the rest of the statements?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -88,3 +88,58 @@ Every bug fix should be accompanied with a unit test, so that we can prevent reg
 ### Documentation, typos, ...
 
 They are mostly welcome!
+
+## Operator SDK
```
Comment (objectiser) [inline CONTRIBUTING.md:92]: Why `Operator SDK` - isn't this section about using the operator via OLM?

File: config/manifests/bases/opentelemetry-operator.clusterserviceversion.yaml [hunk 4]
```diff
@@ -4,63 +4,39 @@ metadata:
...
     alm-examples: '[]'
     capabilities: Basic Install
-    operators.operatorframework.io/builder: operator-sdk-v0.19.0
-    operators.operatorframework.io/project_layout: go.kubebuilder.io/v2
+    categories: Logging & Tracing
+    certified: "false"
+    containerImage: quay.io/opentelemetry/opentelemetry-operator
+    createdAt: "2020-12-16T13:37:00+00:00"
+    description: Provides the OpenTelemetry components, including the Collector
+    operators.operatorframework.io/builder: operator-sdk-v1.2.0
```
Comment (objectiser) [inline config/manifests/bases/opentelemetry-operator.clusterserviceversion.yaml:12]: Is there a way to automatically update these when the sdk version changes? Or does this need to be added to some maintainers documentation somewhere?

File: config/manifests/bases/opentelemetry-operator.clusterserviceversion.yaml [hunk 5]
```diff
@@ -4,63 +4,39 @@ metadata:
...
   name: opentelemetry-operator.vX.Y.Z
   namespace: placeholder
 spec:
   apiservicedefinitions: {}
   customresourcedefinitions:
     owned:
-    - description: OpenTelemetryCollector is the Schema for the opentelemetrycollectors
-        API
+    - description: OpenTelemetryCollector is the Schema for the opentelemetrycollectors API.
+      displayName: Open Telemetry Collector
```
Comment (objectiser) [inline config/manifests/bases/opentelemetry-operator.clusterserviceversion.yaml:23]: Think should be `OpenTelemetry Collector`.

File: CONTRIBUTING.md (hunk 1 above)
Comment (jpkrohling) [reply CONTRIBUTING.md:98]: The PR for 0.17.1 hadn't been merged, but it is now. Will change this PR to reflect that.

File: CONTRIBUTING.md (hunk 2 above)
Comment (jpkrohling) [reply CONTRIBUTING.md:99]: I'll rewrite this section now that we have official confirmation that this is the right path.

File: config/manifests/bases/opentelemetry-operator.clusterserviceversion.yaml (hunk 4 above)
Comment (jpkrohling) [reply config/manifests/bases/opentelemetry-operator.clusterserviceversion.yaml:12]: A new version of the SDK has just been released. I'll make sure to document it in a follow-up PR.

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -89,57 +89,65 @@ Every bug fix should be accompanied with a unit test, so that we can prevent reg
...
+To install our operator, create a `CatalogSource` for our index image, wait for OLM to synchronize and finally create a `Subscription`. Make sure to replace `${USER}` with your username and `${VERSION}` with the version used in the previous step. The namespace for both should be `operators` on Kubernetes, while `openshift-operators` should be used for OpenShift.
+
 ```
 kubectl apply -f - <<EOF
 apiVersion: operators.coreos.com/v1alpha1
 kind: CatalogSource
 metadata:
   name: opentelemetry-operator-manifests
-  namespace: openshift-operators # or olm for Kubernetes
+  namespace: openshift-operators
```
Comment (objectiser) [inline CONTRIBUTING.md:132]: nit: should it be `operators` by default - i.e. by default focus on the base platform (k8s)

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -89,57 +89,65 @@ Every bug fix should be accompanied with a unit test, so that we can prevent reg
...
 
-Install the operator
+When using OpenShift, OLM is already installed.
+
+### Create the bundle and related images
+
+The following commands will generate a bundle under `bundle/` and build an image with its contents. It will then generate and publish an index image with the [Operator Package Manager (OPM)](https://github.com/operator-framework/operator-registry/blob/master/docs/design/opm-tooling.md#opm)
+
+```
+export VERSION=0.17.1
```
Comment (objectiser) [inline CONTRIBUTING.md:115]: nit: to avoid constant updates, should this just be `export VERSION=<required/latest version>

File: RELEASE.md [hunk 8]
```diff
@@ -7,6 +7,9 @@ Steps to release a new version of the OpenTelemetry Operator:
 1. Check the OpenTelemetry Collector's changelog and ensure migration steps are present in `pkg/collector/upgrade`
 1. Once the changes above are merged and available in `master`, tag it with the desired version, prefixed with `v`: `v0.3.0`
 1. The GitHub Workflow will take it from here, creating a GitHub release with the generated artifacts (manifests) and publishing the images
+1. After the release, generate a new OLM bundle (`make bundle`) and create two PRs against the [Operator Hub Community Operators repository](https://github.com/operator-framework/community-operators):
+   1. one for the `community-operators` directory, used by OpenShift. Example: [`operator-framework/community-operators#2878`](operator-framework/community-operators/pull/2878)
+   1. one for the `upstream-community-operators`, used by OLM on Kubernetes. Example: [`operator-framework/community-operators#2880`](operator-framework/community-operators/pull/2880)
```
Comment (objectiser) [inline RELEASE.md:12]: nit: list the k8s line above openshift

**Review by objectiser — COMMENTED** [review]
LGTM - just a couple of nits to consider.

File: CONTRIBUTING.md (hunk 7 above)
Comment (jpkrohling) [reply CONTRIBUTING.md]: I set it to `x.y.z`

**Comment by jpkrohling** [comment]
PR updated

**Review by objectiser — APPROVED** [review]

**Pull request merged**